src/Microsoft.Framework.Configuration/Constants.cs

```diff
@@ -5,6 +5,6 @@ namespace Microsoft.Framework.Configuration
 {
     public static class Constants
     {
-        public static readonly string KeyDelimiter = ":";
+        public static readonly string[] KeyDelimiter = new[] { ":" };
```

Comment:
Because there are some places uses ```StringSplitOptions.RemoveEmptyEntities```, so instead of new up an object in each call, defining ```KeyDelemiter``` as ```string[]``` may suitable in this case